crypto/x509/v3_purp.c

```diff
@@ -419,12 +419,12 @@ int ossl_x509v3_cache_extensions(X509 *x)
         return (x->ex_flags & EXFLAG_INVALID) == 0;
     }
 
+    ERR_set_mark();
+
     /* Cache the SHA1 digest of the cert */
     if (!X509_digest(x, EVP_sha1(), x->sha1_hash, NULL))
```

Comment:
Shouldn't we also remove call to `EVP_sha1` here?

Reply:
Maybe, but there are also many other places where `EVP_sha1()` is still used,
and if this should be changed, it is not in scope of this PR.